db_go18.go

```diff
@@ -0,0 +1,43 @@
+// Copyright 2017 Square Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build go1.8
+
+package squalor
+
+import (
+	"context"
+	"database/sql"
+)
+
+// executor exposes the sql.DB and sql.Tx functions so that it can be used
+// on internal functions that need to be agnostic to the underlying object.
+type executor interface {
+	ExecContext(ctx context.Context, query string, args ...interface{}) (sql.Result, error)
+	QueryContext(ctx context.Context, query string, args ...interface{}) (*sql.Rows, error)
+}
+
+func exec(ec ExecutorContext, query string, args ...interface{}) (sql.Result, error) {
```

Comment:
I think this looks a lot cleaner after the last round of changes. Thanks!

Now, it's clean enough for me to wonder, why do you pass in an `ExecutorContext` here? What you really want is an `executor`, and you have to call this type-switching helper function to turn the former into the latter. Why not just pass in the `executor` needed here?

The only barrier I see is that you call `GetContext()` inside this method. What if instead, the context were passed in to `exec()` etc.? Perhaps you don't need a helper to know which field to extract from which type, you just have the callers provide what is needed directly.

Reply:
You're right, the helper isn't needed. The only slightly weird thing is that this introduces an unused context in `db_go17.go`, but I do think it's cleaner regardless.